src/go/configinfo/service_info.go

```diff
@@ -609,31 +624,45 @@ func (s *ServiceInfo) extractApiKeyLocations(method *methodInfo, parameters []*c
 }
 
 func (s *ServiceInfo) processTypes() error {
-	// Create snake name to JSON name mapping (and validate against duplicates).
-	snakeToJson := make(map[string]string)
 
+	// Convert into map by type name for easy lookup.
+	typesByTypeName := make(map[string]*typepb.Type)
 	for _, t := range s.ServiceConfig().GetTypes() {
-		for _, f := range t.GetFields() {
-			if strings.ContainsRune(f.GetName(), '_') {
+		typesByTypeName[t.Name] = t
+	}
 
-				if prevJsonName, ok := snakeToJson[f.GetName()]; ok {
-					if prevJsonName != f.GetJsonName() {
+	// For each operation, lookup the request type.
+	for operation, requestTypeName := range s.RequestTypeNames {
+		requestType, ok := typesByTypeName[requestTypeName]
+		if !ok {
+			return fmt.Errorf("for operation (%v): could not find type with name (%v)", operation, requestTypeName)
```

Comment:
maybe we should not return error here.  just continue
1) it is easier for our fake service config data, so they don't need to create a type for each method
2) may have problem with grpc,  its type may be embedded, such as foo.bar.  But I am not sure,  maybe for a request type, has to be top level, could not be embeded.

Reply:
Sure, I'm ok with either way. I removed the error, but some comments: 
- 2 should not be an issue, because we will never do mapping for gRPC. 
- 1 makes sense, though I already edited all the affected tests.